lib/salesforce_ar_sync/salesforce_sync.rb

```diff
@@ -180,19 +181,20 @@ def is_boolean?(attribute)
 
     def salesforce_create_object(attributes)
       attributes[self.class.salesforce_web_id_attribute_name.to_s] = id if self.class.salesforce_sync_web_id? && !new_record?
-      result = SF_CLIENT.http_post("/services/data/v#{SF_CLIENT.version}/sobjects/#{salesforce_object_name}", format_attributes(attributes).to_json)
-      self.salesforce_id = JSON.parse(result.body)['id']
+      salesforce_id = SF_CLIENT.create!(salesforce_object_name, format_attributes(attributes))
+      self.salesforce_id = salesforce_id
       @exists_in_salesforce = true
     end
 
     def salesforce_update_object(attributes)
       attributes[self.class.salesforce_web_id_attribute_name.to_s] = id if self.class.salesforce_sync_web_id? && !new_record?
-      SF_CLIENT.http_patch("/services/data/v#{SF_CLIENT.version}/sobjects/#{salesforce_object_name}/#{salesforce_id}", format_attributes(attributes).to_json)
+
+      SF_CLIENT.update!(salesforce_object_name, format_attributes(attributes).merge(Id: salesforce_id))
```

Comment:
There's one other place where this method gets called (inside the sync delayed_job itself) which doesn't have a rescue:

https://github.com/InfoTech/salesforce_ar_sync/blob/master/lib/salesforce_ar_sync/salesforce_object_sync.rb#L8

Is it possible with this .update! this job could fail and requeue for a while?  Maybe that's not a big deal, just figured it might be worth mentioning because it looks like the old http_patch method couldn't have thrown errors, and now it can.